lab_4_fill_words_by_ngrams/main.py

```diff
@@ -28,6 +32,19 @@ def _tokenize(self, text: str) -> tuple[str, ...]:  # type: ignore
         Raises:
             ValueError: In case of inappropriate type input argument or if input argument is empty.
         """
+        if not (isinstance(text, str) and text):
+            raise ValueError
+        punctuation_to_replace = '.!?'
+        tokens = []
+        for word in text.lower().split():
+            if word[-1] in punctuation_to_replace:
+                tokens.extend([word[:-1], self._end_of_word_token])
```

Comment:
no cleaning for chunks with punctuation ending?

Reply:
I tried to fix it